Assets/MixedRealityToolkit-Preview/LightEstimation/Scripts/LightCapture.cs

```diff
@@ -0,0 +1,256 @@
+			probe = probeObj.GetComponent<ReflectionProbe>();
+			
+			probe.size          = Vector3.one * 10000;
+			probe.boxProjection = false;
+		}
+
+		// Same with a camera object
+		if (cameraOrientation == null)
+			cameraOrientation = Camera.main.transform;
+
+		// And check for a directional light in the scene
+		if (useDirectionalLight && directionalLight == null)
+		{
+			Light[] lights = FindObjectsOfType<Light>();
+			for (int i = 0; i < lights.Length; i++)
+			{
+				if (lights[i].type == LightType.Directional)
+				{
+					directionalLight = lights[i];
+					break;
+				}
+			}
+			if (directionalLight == null)
+			{
+				GameObject lightObj = new GameObject("_DirectionalLight", typeof(Light));
+				lightObj.transform.SetParent(transform, false);
+				directionalLight = lightObj.GetComponent<Light>();
+				directionalLight.type = LightType.Directional;
+			}
+		}
+
+		// Save initial settings in case we wish to restore them
+		startSky = RenderSettings.skybox;
+	}
+	private void OnDisable()
+	{
+		// Restore and render a default probe
+		if (probe != null && probe.isActiveAndEnabled)
+		{
+			probe.mode        = UnityEngine.Rendering.ReflectionProbeMode.Realtime;
+			probe.refreshMode = UnityEngine.Rendering.ReflectionProbeRefreshMode.ViaScripting;
+			probe.RenderProbe();
+		}
+
+		RenderSettings.skybox = startSky;
+
+		captureCamera.Shutdown();
+	}
+	private void OnEnable()
+	{
+		probe.mode = UnityEngine.Rendering.ReflectionProbeMode.Custom;
+		
+		CameraResolution resolution = new CameraResolution();
+		resolution.nativeResolution = NativeResolutionMode.Smallest;
+		resolution.resize           = ResizeWhen.Never;
+
+		captureCamera.Initialize(true, resolution, ()=>
+		{
+			if (map == null)
+			{
+				map = new CubeMapper();
+				map.Create(captureCamera.FieldOfView * stampFovMultiplier, mapResolution);
+				map.StampExpireDistance = stampExpireDistance;
+			}
+			probe.customBakedTexture = map.Map;
+			RenderSettings.skybox = map.SkyMaterial;
+			RenderSettings.ambientMode = AmbientMode.Skybox;
+		});
+	}
+	private void OnValidate()
+	{
+		if (map != null)
+			map.StampExpireDistance = stampExpireDistance;
+	}
+
+	private void Update ()
+	{
+		// ditch out if we already have our first stamp
+		if ((stampCount > 0 && singleStampOnly) )
+			return;
+
+		// check the cache to see if our current orientation would benefit from a new stamp
+		if (captureCamera.IsReady && !captureCamera.IsRequestingImage)
+		{
+			if (!map.IsCached(cameraOrientation.position, cameraOrientation.rotation))
+			{
+				captureCamera.RequestImage(OnReceivedImage);
+			}
+		}
+
+		// If we have a target light rotation to get to, start lerping! 
+		if (lightStartTime > 0)
+		{
+			float t =  Mathf.Clamp01( (Time.time - lightStartTime) / lightTargetDuration );
+			if (t >= 1)
+				lightStartTime = 0;
+			
+			// This is a cheap cubic in/out easing function, so we aren't doing this linear (ew)
+			t = t<.5f ? 4*t*t*t : (t-1)*(2*t-2)*(2*t-2)+1; 
```

Comment:
Nit: This was cool to watch on device. You could use a AnimationCurve property to allow the user more control and eliminate this easing function.

Reply:
I had thought about using an AnimationCurve here! I ultimately decided that the ability to customize this to such a degree is probably overkill, it's way too easy to make a bad curve, and pretty hard to make a curve better than this one!